contracts/contracts/vault/OETHVaultAdmin.sol

```diff
@@ -1,12 +1,102 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IStrategy } from "../interfaces/IStrategy.sol";
+import { IVault } from "../interfaces/IVault.sol";
 import { VaultAdmin } from "./VaultAdmin.sol";
 
 /**
  * @title OETH VaultAdmin Contract
  * @author Origin Protocol Inc
  */
 contract OETHVaultAdmin is VaultAdmin {
+    using SafeERC20 for IERC20;
+
+    address public immutable weth;
+
+    constructor(address _weth) {
+        weth = _weth;
+    }
+
+    /// @dev Simplified version of the deposit function as WETH is the only supported asset.
+    function _depositToStrategy(
+        address _strategyToAddress,
+        address[] calldata _assets,
+        uint256[] calldata _amounts
+    ) internal override {
+        require(
+            strategies[_strategyToAddress].isSupported,
+            "Invalid to Strategy"
+        );
+        require(
+            _assets.length == 1 && _amounts.length == 1 && _assets[0] == weth,
+            "Only WETH is supported"
+        );
+
+        // Check the there is enough WETH to transfer once the WETH reserved for the withdrawal queue is accounted for
+        require(_amounts[0] <= _wethAvailable(), "Not enough WETH available");
+
+        // Send required amount of funds to the strategy
+        IERC20(weth).safeTransfer(_strategyToAddress, _amounts[0]);
+
+        // Deposit all the funds that have been sent to the strategy
+        IStrategy(_strategyToAddress).depositAll();
+    }
+
+    function _withdrawFromStrategy(
+        address _recipient,
+        address _strategyFromAddress,
+        address[] calldata _assets,
+        uint256[] calldata _amounts
+    ) internal override {
+        super._withdrawFromStrategy(
+            _recipient,
+            _strategyFromAddress,
+            _assets,
+            _amounts
+        );
+
+        IVault(address(this)).addWithdrawalQueueLiquidity();
+    }
+
+    function _withdrawAllFromStrategy(address _strategyAddr) internal override {
+        super._withdrawAllFromStrategy(_strategyAddr);
+
+        IVault(address(this)).addWithdrawalQueueLiquidity();
+    }
+
+    function _withdrawAllFromStrategies() internal override {
+        super._withdrawAllFromStrategies();
+
+        IVault(address(this)).addWithdrawalQueueLiquidity();
+    }
+
+    /// @dev Calculates the amount of WETH in the Vault that is not reserved for the withdrawal queue.
+    function _wethAvailable() internal view returns (uint256 wethAvailable) {
```

Comment:
🟠 Our two WETH available functions do not match.

Reply:
Fixed. I think this is a good argument to getting rid of the duplicate code